test/app-luatest/override_misc_test.lua

```diff
@@ -52,3 +52,21 @@ g.test_initializing = function(g)
     local exp = expected_output('socket')
     t.assert_equals(res, exp)
 end
+
+g.test_no_package_loaded = function(g)
+    local load_cnt = 0
+    local foo = {whoami = 'foo'}
+    package.preload.foo = function()
+        local loaders = require('internal.loaders')
+        load_cnt = load_cnt + 1
+        if load_cnt < 3 then
+            loaders.no_package_loaded.foo = true
+        end
+        return foo
+    end
+    for i = 1, 10 do
```

Comment:
`i` is unused, see luacheck warnings.

Reply:
Also, can we use `LOAD_LIMIT + 1` here to be more consistent?

Reply:
No problem. Added `LOAD_ATTEMPTS` as `LOAD_LIMIT + 1`.